### DISCUSSION TITLE
https://docsaid.org/docs/docaligner/model_arch

### DISCUSSION THREAD

**Discussion by giscus** [DocsaidLab/website#1] [Announcements]
# https://docsaid.org/docs/docaligner/model_arch

我們參考過去的研究文獻，首先考慮了點回歸模型。

https://docsaid.org/docs/docaligner/model_arch

**Comment by cactusgame** [comment]
你好，首先非常感谢分享的文章。

关于”解决放大误差“时，使用”计算Contour质心”的办法，感觉效果也不是很好。在Heatmap的基础上，先确定一个小范围的图形，在这个小范围图形上做回归（图片已经不大了，所以速度不会很慢），效果会不会更好一些？

**Reply by zephyr-sh** [reply]
我覺得可以！

這讓我想到或許可以直接取出Heatmap上亮區附近的特徵圖（或原圖），直接串一個 branch 出來做回歸，也算是另類的 anchor 想法。

謝謝你的分享，我再安排時間嘗試看看。

**Reply by cactusgame** [reply]
另外，想请教一下。从Github上看，DocAligner似乎只开放了infer相关的代码，请问训练相关代码未来有计划开源么？

**Reply by zephyr-sh** [reply]
謝謝詢問，我目前沒有開源訓練模組的規劃。

如果有需要的話，我建議可以參考這幾個地方：

- **訓練環境建置**：可參考 DocClassifier 專案，那個專案有開放完整的訓練模組，我所有訓練模組都是走相同的流程。
- **backbone**：這個直接去調用 timm。
- **neck**：這個在 docsaidkit 專案內。
- **head**：這是唯一需要自己實作的地方，但架構設計有寫在技術文件內。
- **dataset**：都是公開資料集，也有寫在技術文件內。

**Reply by cactusgame** [reply]
非常感谢您的分享！


> 謝謝詢問，我目前沒有開源訓練模組的規劃。
> 
> 如果有需要的話，我建議可以參考這幾個地方：
> 
> * **訓練環境建置**：可參考 DocClassifier 專案，那個專案有開放完整的訓練模組，我所有訓練模組都是走相同的流程。
> * **backbone**：這個直接去調用 timm。
> * **neck**：這個在 docsaidkit 專案內。
> * **head**：這是唯一需要自己實作的地方，但架構設計有寫在技術文件內。
> * **dataset**：都是公開資料集，也有寫在技術文件內。
